build-tools/packages/bundle-size-tools/src/ADO/AdoArtifactFileProvider.ts

```diff
@@ -43,11 +37,21 @@ export async function getZipObjectFromArtifact(
 ): Promise<JSZip> {
 	const buildApi = await adoConnection.getBuildApi();
 
+	// IMPORTANT
+	// azure-devops-node-api has a known (and seemingly ignored) issue that makes it not work if we need to download
+	// artifacts created by the new PublishPipelineArtifact task (it worked for the old PublishBuildArtifacts task).
+	// The move to 1ES pipeline templates forced our hand to use the new task (through an 1ES-provided task template).
+	// The workaround is to override the createAcceptHeader function when making the request to download the artifact.
+	// See https://github.com/microsoft/azure-devops-node-api/issues/432 for more details
+	const originalCreateAcceptHeader = buildApi.createAcceptHeader;
+	buildApi.createAcceptHeader = (type: string): string => type;
```

Comment:
Nit: The comment in the GH issue is helpful for understanding what the workaround does and why it's necessary.  Perhaps replicate it here?  (slightly sanitized).

```
// ADO's APIs try to download pipeline artifacts using an API version that isn't supported
// by the artifact download endpoint.
// One way of getting around that is by temporarily removing the API version to force it to use
// a supported version.
// See: https://github.com/microsoft/azure-devops-node-api/issues/432
```

The historical information (used to work, 1ES forced our hand, etc.) might be better expressed in the PR comments, since it's explaining the motivation for the delta, not the motivation for the code.